src/components/atoms/Button/Button.tsx

```diff
@@ -0,0 +1,21 @@
+import { Button as AntDesignButton } from "antd";
+import {
+  ButtonType as AntDesignButtonType,
+  ButtonProps as AntDesignButtonProps,
+} from "antd/lib/button";
+
+interface ButtonProps extends AntDesignButtonProps {
+  type?: AntDesignButtonType;
+  buttonText?: string;
+  onClick?: () => void;
+}
```

Comment:
`type` and `onClick` is already defined in `AntDesignButtonProps` so no need to defined them again. Also let's remove `buttonText` so you just pass in text as children, then it is easier to allow custom components to be passed in if necessary. Also let's add an optional `to` prop so we can wrap the button in a link
```
import { Button as AntDesignButton } from "antd";
import { ButtonProps as AntDesignButtonProps } from "antd/lib/button";
import { Link, LinkProps } from "react-router-dom";

interface ButtonProps extends Omit<AntDesignButtonProps, "href"> {
  /**
   * Turn button into link (optional)
   */
  to?: LinkProps["to"];
}

const Button = ({ to, ...buttonProps }: ButtonProps) => {
  const buttonContent = <AntDesignButton {...buttonProps} />;

  if (to) {
    return <Link to={to}>{buttonContent}</Link>;
  }

  return buttonContent;
};

export default Button;
```

Basically look at the Kaylo button as inspiration, would be nice to also add the logic `isURL` so we can show a `<a>` if link is external